addons/portal/views/portal_templates.xml

```diff
@@ -833,10 +731,330 @@
             <div class="o_portal_my_details">
                 <div t-field="user_id.partner_id" t-options='{"widget": "contact", "fields": ["email", "phone", "address"]}'/>
             </div>
-            <a role="button" href="/my/account" class="btn btn-link p-0 mt-3"><i class="fa fa-pencil"/> Edit information</a>
+            <a t-attf-href="/portal/address?partner_id={{user_id.partner_id.id}}&amp;portal_address=True" class="btn btn-link o_edit_info p-0" role="button">
+                <i class="fa fa-pencil"/> Edit information
+            </a>
             <div class="o_my_contact mt-5" t-if="sales_user">
                 <t t-call="portal.portal_contact"/>
             </div>
         </div>
     </template>
+
+    <template id="portal_my_addresses">
+        <t t-set="additional_title" t-value="'My Addresses'"/>
+        <t t-call="portal.portal_layout">
+            <div t-if="request.params.get('error')" class="alert alert-warning" role="alert" groups="account.group_delivery_invoice_address">
+                <h4 class="alert-heading">You are not allowed to delete this address!</h4>
+                <p>
+                    You cannot archive contacts linked to an active user.You first need to archive their associated user.Linked active users : test
+                </p>
+            </div>
+            <h5 class="mt-4">Billing address</h5>
+            <div id="address_checkout_billing" class="one_kanban all_billing_address row">
+                <t t-if="partner">
+                    <div class="col-lg-4 mb-3" style="min-height: 200px">
+                        <t t-call="portal.address_card">
+                            <t t-set="address" t-value="partner"/>
+                            <t t-set="is_invoice" t-value="True"/>
+                            <t t-set="selected" t-value="len(partner) == 1 and partner.is_default_billing_address"/>
+                        </t>
+                    </div>
+                    <t t-foreach="partner.child_ids.filtered(lambda c: c.type == 'invoice' or c.type == 'other')" t-as="billing_address">
+                        <div class="col-lg-4 mb-3" style="min-height: 200px">
+                            <t t-call="portal.address_card">
+                                <t t-set="is_invoice" t-value="True"/>
+                                <t t-set="address" t-value="billing_address"/>
+                                <t t-set="selected" t-value="billing_address.is_default_billing_address"/>
+                            </t>
+                        </div>
+                    </t>
+                    <div class="col-lg-4 mb-3" style="min-height: 200px">
+                        <a role="button" href="/portal/address?address_type=other&amp;portal_address=True" class="d-flex flex-column align-items-center justify-content-center h-100 p-4 mx-auto rounded text-decoration-none" style="border: 1px dashed">
```

Comment:
I guess addresses need to be other if they're not the default?

It's more of a functional comment if so, but it seems odd to always add the address to both lists. I feel like ideally adding a billing address should not necessarily add it to my list of delivery addresses. But maybe that's difficult to do..